src/nimlsp.nim

```diff
@@ -145,9 +147,20 @@ proc getProjectFile(file: string): string =
       fileExists(path / current.addFileExt(".nims"))) and certainty <= Cfg:
       result = path / current.addFileExt(".nim")
       certainty = Cfg
-    if fileExists(path / current.addFileExt(".nimble")) and certainty <= Nimble:
-      # Read the .nimble file and find the project file
-      discard
+    for file in walkDirRec(path):
+      if file.endsWith(".nimble") and certainty <= Nimble:
+        # Read the .nimble file and find the project file
+        let content = readFile(file)
+        let lines = content.splitLines()
+        let (dir, fname, ext) = file.splitFile()
+        for line in lines:
+          if scanf(line,"""$ssrcDir$s=$s"$w"$s""",srcDir):
```

Comment:
This is technically a hack as Nimble files are NimScript and can't be trivially parsed like this. But I will accept this solution until we have something better, but please leave a `TODO` comment to point this out.

Reply:
better to invoke it on `nimble dump` output.

Reply:
I take it more carefully now, it will ignore case and triple quote also work.

Reply:
Doing this with `nimble dump` is indeed a great idea SolitudeSF! This should be implemented instead of trying to read the file (just make sure that `nimble` is executable).